cdap-unit-test/src/test/java/co/cask/cdap/test/artifacts/plugins/ToStringPlugin.java

```diff
@@ -27,7 +28,9 @@
  */
 @Plugin(type = "t1")
 @Name("n1")
-public class ToStringPlugin {
+public class ToStringPlugin implements Serializable {
+
+  private static final long serialVersionUID = -4818590442766677617L;
```

Comment:
Depends on how it is being used. It not used through closure, then no. Currently there are two ways of using plugin in Spark:

```java
public void run(SparkContext context) {
  final PluginContenxt pluginContext = context.getPluginContext();
  final MyPlugin p = pluginContext.newPluginInstance("plugin");
  ...
  someRDD.map(new Function<String, String>() {
    public String call(String str) {
      // MyPlugin needs to be serializable
      p.doSomething();
      // MyPlugin2 doesn't need to be serializable
      MyPlugin2 p2 = pluginContext.newPluginInstance("plugin2");
    }
  });
}

```